File: 0056-merge-intervals/0056-merge-intervals.java
class Solution {
    public int[][] merge(int[][] intervals) {

        List<int[]> result = new ArrayList<>();
        int i=1;

        Arrays.sort(intervals, (i1, i2) -> Integer.compare(i1[0], i2[0]));

        int[] newInterval = intervals[0];
        while(i < intervals.length){
            if(newInterval[1] < intervals[i][0]){
                result.add(newInterval);
                newInterval = intervals[i];
            } {
                newInterval[0] = Math.min(newInterval[0], intervals[i][0]);
                newInterval[1] = Math.max(newInterval[1], intervals[i][1]);            
            }
            i++;
        }       
        result.add(newInterval); 

        return result.toArray(new int[result.size()][]); 
    }
}

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        
        intervals.sort()
        result = []

        minInterval, maxInterval = intervals[0][0], intervals[0][1]
        for i in range(1, len(intervals)):
            if maxInterval >= intervals[i][0]:
                minInterval = min(minInterval, intervals[i][0])
                maxInterval = max(maxInterval, intervals[i][1])                
            else:
                result.append([minInterval, maxInterval])
                minInterval, maxInterval = intervals[i][0], intervals[i][1]
        result.append([minInterval, maxInterval])        

        return result